aidial_assistant/json_stream/json_array.py

```diff
@@ -28,50 +29,40 @@ def type(self) -> str:
     def token() -> str:
         return "["
 
-    @override
-    def __aiter__(self) -> AsyncIterator[JsonNode]:
-        return self
-
-    @override
-    async def __anext__(self) -> JsonNode:
-        result = await self.listener.get()
-        if result is None:
-            raise StopAsyncIteration
-
-        self.array.append(result)
-        return result
-
-    @override
-    async def parse(
-        self, stream: CharacterStream, dependency_resolver: NodeResolver
-    ):
-        normalised_stream = JsonNormalizer(stream)
-        char = await anext(normalised_stream)
-        self._char_position = stream.char_position
-        if not char == JsonArray.token():
-            raise unexpected_symbol_error(char, stream.char_position)
-
-        separate = False
-        while True:
-            char = await normalised_stream.apeek()
-            if char == "]":
-                await anext(normalised_stream)
-                break
-
-            if char == ",":
-                if not separate:
-                    raise unexpected_symbol_error(char, stream.char_position)
-
-                await anext(normalised_stream)
-                separate = False
-            else:
-                value = await dependency_resolver.resolve(stream)
-                await self.listener.put(value)
-                if isinstance(value, ComplexNode):
-                    await value.parse(stream, dependency_resolver)
-                separate = True
-
-        await self.listener.put(None)
+    @staticmethod
+    async def read(
+        stream: CharacterStream, dependency_resolver: NodeResolver
```

Comment:
`dependency_resolver` sounds really strange, what are those dependencies? The notion of 'dependency' is nowhere else to be found in the code. Simply `parser` is telling enough.